sc-kpm/sc-agents-common/CMakeLists.txt

```diff
@@ -13,7 +13,12 @@ target_link_libraries(sc-agents-common
 )
 target_include_directories(sc-agents-common
     PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src
-    PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include
+    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    PUBLIC $<INSTALL_INTERFACE:include>
+)
```

Comment:
I'll add my 5 cents: installation process in this context is us exporting CMake targets for devs to consume sc-machine as a library, so basically `INSTALL_INTERFACE` is needed for cases when sc-machine is installed as a Conan package or through GitHub Releases archives rather than by building from source